### ISSUE TITLE
Question: why are there new script tags created for every event?

### ISSUE THREAD

**Issue by thomhos** [alinemorelli/react-gtm#3]
It seems that for every event a new script tag is created in the head which reassigns the dataLayer and then pushes the event.

I wonder why this is, is it for SSR compatibility or some other reason?

**Comment by alinemorelli** [comment]
Can you show me how do you implement it?
Do you change dataLayer information for every event or is it reassigning same information?

**Comment by thomhos** [comment]
So basically we do this before bootstrapping the app:

```js
const tagManagerArgs = {
    gtmId: env.GOOGLE_TAG_MANAGER_TOKEN,
    dataLayerName: 'PageDataLayer',
}

TagManager.initialize(tagManagerArgs)
```
Then later on every route change we do this:
```js
const tagManagerArgs = {
    dataLayer: {
        event: 'VirtualPageview',
            virtualPageURL: window.location.href,
            virtualPageTitle: document.title,
         },
    dataLayerName: 'PageDataLayer',
}
TagManager.dataLayer(tagManagerArgs)
```

We end up with a new script tag for each time the above is executed. I assume it's by design, but mainly wondering why :)

**Comment by Vadorequest** [comment]
@alinemorelli I don't get why his implementation would change this behaviour. Since it's the dataLayer method that creates a new script every time it's called

https://github.com/alinemorelli/react-gtm/blob/master/src/TagManager.js#L46

I don't think that's really a good thing.

**Comment by rafaelcavalcante** [comment]
Any update on this subject?

**Comment by richardblondet** [comment]
https://github.com/alinemorelli/react-gtm/blob/master/src/TagManager.js#L47 the dataLayer appears to be initialized upon gtag first run. It serves as a page load, collects all the parameters and fires.

it is not elegant but does the work

**Comment by alinemorelli** [comment]
Fixed on #22
